### PR TITLE
Bulk update (C#6.0 and resource strings)

### PR DESCRIPTION
**Pull request by maartenba**
This PR fixes:

* Convert gallery to C#6.0 #2815
* Move gallery user-visible strings (e.g. in exceptions) to resources #2816

### CODE REVIEW

File: .vs/config/applicationhost.config [hunk 1]
```diff
@@ -155,7 +155,7 @@
         <sites>
             <site name="NuGet Gallery (nuget.localtest.me)" id="2">
                 <application path="/">
-                    <virtualDirectory path="/" physicalPath="..\..\src\NuGetGallery" />
+                    <virtualDirectory path="/" physicalPath="D:\Projects\Git\NuGetGallery\src\NuGetGallery" />
```
Comment (yishaigalatzer) [inline .vs/config/applicationhost.config]: This looks like a bug, it will only work on the specific machine the project was cloned to

File: src/NuGetGallery.Core/Auditing/CloudAuditingService.cs [hunk 2]
```diff
@@ -102,9 +102,8 @@ public static Task<AuditActor> AspNetActorThunk()
             }
             catch (StorageException ex)
             {
-                if(ex.RequestInformation != null && 
-                    ex.RequestInformation.ExtendedErrorInformation != null &&
-                    ex.RequestInformation.ExtendedErrorInformation.ErrorCode == BlobErrorCodeStrings.ContainerNotFound)
+                if(ex.RequestInformation?.ExtendedErrorInformation != null
```
Comment (yishaigalatzer) [inline src/NuGetGallery.Core/Auditing/CloudAuditingService.cs]: nit: I think you can ?. all the way .? :)

File: src/NuGetGallery.Core/packages.config [hunk 3]
```diff
@@ -8,7 +8,7 @@
   <package id="Microsoft.WindowsAzure.ConfigurationManager" version="3.1.0" targetFramework="net45" />
   <package id="Newtonsoft.Json" version="6.0.4" targetFramework="net45" />
   <package id="NuGet.Core" version="2.8.6" targetFramework="net45" />
-  <package id="NuGet.Services.Build" version="3.0.9" targetFramework="net45" developmentDependency="true" />
+  <package id="NuGet.Services.Build" version="3.0.13" targetFramework="net452" developmentDependency="true" />
```
Comment (yishaigalatzer) [inline src/NuGetGallery.Core/packages.config]: why did the TFM change here?

File: .vs/config/applicationhost.config (hunk 1 above)
Comment (maartenba) [inline .vs/config/applicationhost.config]: Dev branch will ignore the file, vs regenerates it anyway. Will remove it in here as well.

File: src/NuGetGallery.Core/packages.config (hunk 3 above)
Comment (maartenba) [inline src/NuGetGallery.Core/packages.config]: No idea, but it should not matter as the package only adds a few .props and .targets.

File: src/NuGetGallery.Core/Auditing/CloudAuditingService.cs (hunk 2 above)
Comment (maartenba) [inline src/NuGetGallery.Core/Auditing/CloudAuditingService.cs]: Fixed.

File: tests/NuGetGallery.Facts/SearchClient/RetryingHttpClientWrapperFacts.cs [hunk 4]
```diff
@@ -35,7 +35,7 @@ public class RetryingHttpClientWrapperFacts
         private static readonly Uri ValidUri2 = new Uri("http://www.bing.com");
         private static readonly Uri InvalidUri1 = new Uri("http://nonexisting.domain.atleast.ihope");
         private static readonly Uri InvalidUri2 = new Uri("http://nonexisting.domain.atleast.ihope/foo");
-        private static readonly Uri InvalidUri3 = new Uri("http://www.nuget.org/packages?q=%22Windows&sortOrder=package-download-count&page=7&prerelease=False");
+        private static readonly Uri InvalidUri3 = new Uri("http://www.nuget.org/com/ibm/mq/com.ibm.mq.soap/7.0.1.10/com.ibm.mq.soap-7.0.1.10");
```
Comment (yishaigalatzer) [inline tests/NuGetGallery.Facts/SearchClient/RetryingHttpClientWrapperFacts.cs]: looks unrelated

File: tests/NuGetGallery.FunctionalTests.Core/Helpers/ClientSdkHelper.cs [hunk 5]
```diff
@@ -68,7 +68,7 @@ public bool CheckIfPackageVersionExistsInSource(string packageId, string version
             SemanticVersion semVersion;
             var success = SemanticVersion.TryParse(version, out semVersion);
             const int interval = 30;
-            const int maxAttempts = 10;
+            const int maxAttempts = 15;
```
Comment (yishaigalatzer) [inline tests/NuGetGallery.FunctionalTests.Core/Helpers/ClientSdkHelper.cs]: looks unrelated, why?

**Comment by yishaigalatzer** [comment]
:shipit: from me, this should PR should have another set of eyes looking at it. Particularly at the $"" changes

File: src/NuGetGallery.Core/packages.config (hunk 3 above)
Comment (xavierdecoster) [inline src/NuGetGallery.Core/packages.config]: I've seen this happening when at some point an upgrade of the project happened from net45 to net452, and no `Update-Package -Reinstall` was run against the project, leaving the packages.config in TFM net45 state. The next update of the package then makes it look as if a TFM switch happened.

File: tests/NuGetGallery.Facts/SearchClient/RetryingHttpClientWrapperFacts.cs (hunk 4 above)
Comment (maartenba) [inline tests/NuGetGallery.Facts/SearchClient/RetryingHttpClientWrapperFacts.cs]: Comes from dev branch, makes the tests all green. Unrelated to this PR indeed and will be rebased out once :shipit:

File: src/NuGetGallery.Core/packages.config (hunk 3 above)
Comment (maartenba) [inline src/NuGetGallery.Core/packages.config]: That explains. Ran an `Update-Package -Reinstall` so they all are matched properly.

File: src/NuGetGallery.Core/Auditing/AuditingService.cs [hunk 6]
```diff
@@ -71,7 +71,8 @@ private class NullAuditingService : AuditingService
         {
             protected override Task<Uri> SaveAuditRecord(string auditData, string resourceType, string filePath, string action, DateTime timestamp)
             {
-                var uriString = string.Format("http://auditing.local/{0}/{1}/{2}-{3}", resourceType, filePath, timestamp.ToString("s"), action.ToLowerInvariant());
+                var uriString =
+                    $"http://auditing.local/{resourceType}/{filePath}/{timestamp.ToString("s")}-{action.ToLowerInvariant()}";
```
Comment (joelverhagen) [inline src/NuGetGallery.Core/Auditing/AuditingService.cs]: nit `{timestamp:s}`

File: src/NuGetGallery.Operations/Infrastructure/JobLogBlob.cs [hunk 7]
```diff
@@ -31,7 +31,7 @@ public JobLogBlob(CloudBlockBlob blob)
             var parsed = BlobNameParser.Match(blob.Name);
             if (!parsed.Success)
             {
-                throw new ArgumentException("Job Log Blob name is invalid Bob! Expected [jobname].[yyyy-MM-dd].json or [jobname].json. Got: " + blob.Name, "blob");
+                throw new ArgumentException(string.Format("Job Log Blob name is invalid Blob! Expected [jobname].[yyyy-MM-dd].json or [jobname].json. Got: {0}", blob.Name), nameof(blob));
```
Comment (joelverhagen) [inline src/NuGetGallery.Operations/Infrastructure/JobLogBlob.cs]: Could use string interpolation.

File: src/NuGetGallery.Operations/Tasks/Database/CreateSqlUserTask.cs [hunk 8]
```diff
@@ -28,7 +28,7 @@ public override void ValidateArguments()
 
             if (String.IsNullOrEmpty(UserName) && CurrentEnvironment != null)
             {
-                UserName = String.Format("{0}-site-{1}", CurrentEnvironment.EnvironmentName, DateTime.UtcNow.ToString("MMMdd-yyyy"));
+                UserName = $"{CurrentEnvironment.EnvironmentName}-site-{DateTime.UtcNow.ToString("MMMdd-yyyy")}";
```
Comment (joelverhagen) [inline src/NuGetGallery.Operations/Tasks/Database/CreateSqlUserTask.cs]: nit `{DateTime.UtcNow:MMMdd-yyyy}`

File: src/NuGetGallery.Operations/Tasks/UploadPackageTask.cs [hunk 9]
```diff
@@ -31,11 +31,7 @@ public override void ExecuteCommand()
 
             var container = client.GetContainerReference("packages");
 
-            var fileName = string.Format(
-                "{0}.{1}{2}",
-                PackageId,
-                PackageVersion,
-                ".nupkg");
+            var fileName = $"{PackageId}.{PackageVersion}{".nupkg"}";
```
Comment (joelverhagen) [inline src/NuGetGallery.Operations/Tasks/UploadPackageTask.cs]: `$"{PackageId}.{PackageVersion}.nupkg"` is simpler

File: src/NuGetGallery/App_Start/Routes.cs [hunk 10]
```diff
@@ -179,6 +179,12 @@ public static void RegisterUIRoutes(RouteCollection routes)
                 new { controller = "Authentication", action = "Authenticate" });
 
             routes.MapRoute(
+                "RegisterAccount",
```
Comment (joelverhagen) [inline src/NuGetGallery/App_Start/Routes.cs]: How is this related?

**Comment by joelverhagen** [comment]
A couple nits. Otherwise looks like great cleanup. :shipit:

File: src/NuGetGallery/App_Start/Routes.cs (hunk 10 above)
Comment (maartenba) [inline src/NuGetGallery/App_Start/Routes.cs]: Seems like a space was removed or something. No biggie.

**Comment by maartenba** [comment]
Cleaned up. Will merge after we ship the NuGet.Core refactoring.

**Comment by maartenba** [comment]
Reopened PR here (just the C# 6.0 changes) - https://github.com/NuGet/NuGetGallery/pull/2846
